core/src/main/java/org/fao/geonet/migration/JsonUISettingDatabaseMigration.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2001-2021 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+package org.fao.geonet.migration;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.lang.StringUtils;
+import org.fao.geonet.DatabaseMigrationTask;
+import org.fao.geonet.constants.Geonet;
+import org.fao.geonet.utils.Log;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Helper class for handling JSON strings for UI Settings table.
+ */
+public abstract class JsonUISettingDatabaseMigration extends DatabaseMigrationTask {
+
+    public void update(Connection connection) throws SQLException {
+        try (PreparedStatement statement = connection.prepareStatement(
+            "SELECT * FROM settings_ui")) {
+            ResultSet rs = statement.executeQuery();
+
+            if (rs.next()) {
+                String currentSettingJson = rs.getString("configuration");
+                if (StringUtils.isNotEmpty(currentSettingJson)) {
+                    Map<String, String> fieldsToUpdate = setUpNewSettingValues();
+
+                    String newSettingJson = insertOrUpdateField(currentSettingJson, fieldsToUpdate);
+                    try (PreparedStatement updateStatement = connection.prepareStatement("UPDATE settings_ui SET "
+                        + "configuration=? WHERE id=?")) {
+                        updateStatement.setString(1, newSettingJson);
+                        updateStatement.setInt(2, rs.getInt("id"));
```

Comment:
Testing it in `3.12.x` with a 3.10.x database it fails since the `settingsUI.id` column is a string instead of an integer:
https://github.com/geonetwork/core-geonetwork/blob/1a61cfd6f5bc407f5054ac026d9a4c366213e094/domain/src/main/java/org/fao/geonet/domain/UiSetting.java#L67-L75

Reply:
Changed to `rs.setString(2, rs.getString("id"))`.